doc/rtd/topics/debugging.rst

```diff
@@ -163,3 +163,103 @@ commandline:
+   **-proposed** APT pocket (e.g. **bionic-proposed**)
+ * Upgrade cloud-init and attempt a clean run of cloud-init to assert the new
+   version of cloud-init works properly the specific platform and Ubuntu series
+ * Check for tracebacks or errors in behavior
+
+
+Manual SRU verification procedure
+---------------------------------
+Below are steps that can manually test a pre-release version of cloud-init
+from **-proposed**
+
+.. note::
+    For each SRU, upstream manually validates cloud-init upgrades for **Ubuntu**
+    on these platforms: **Amazon Ec2, Azure, GCE, OpenStack, Oracle,
+    Softlayer (IBM), NoCloud (LXD&KVM)**
+
+1. Launch a VM on your favorite platform, providing the cloud-config
+   user-data and replacing `<YOUR_LAUNCHPAD_USERNAME>` with your username:
+
+.. code-block:: yaml
+
+    ## template: jinja
+    #cloud-config
+    ssh_import_id: [<YOUR_LAUNCHPAD_USERNAME>]
+    hostname: SRU-worked-{{v1.cloud_name}}
+
+2. Wait for current cloud-init to complete, replace `<YOUR_VM_IP>` with the IP
+   address of the VM that you launched in step 1:
+
+.. code:: bash
+
+    CI_VM_IP=<YOUR_VM_IP>
+    # Make note of the datasource cloud-init detected in --long output.
+    # In step 5, confirm the same datasource got detected after upgrade.
+    ssh ubuntu@$CI_VM_IP -- cloud-init status --wait --long
+
+3. Set up the **-proposed** pocket on your VM and upgrade to the **-proposed**
+   cloud-init
+
+.. code:: bash
+
+    # Create a script that will add the -proposed pocket to APT's sources
+    # and install cloud-init from that pocket
+    cat > setup_proposed.sh <<EOF
+    #/bin/bash
+    mirror=http://archive.ubuntu.com/ubuntu
+    echo deb \$mirror \$(lsb_release -sc)-proposed main | tee \
+        /etc/apt/sources.list.d/proposed.list
+    apt-get update -q
+    apt-get install -qy cloud-init
+    EOF
+
+    scp setup_proposed.sh ubuntu@$CI_VM_IP:.
+    ssh ubuntu@$CI_VM_IP -- sudo bash setup_proposed.sh
+
+4. Change hostname, clean cloud-init's state, and reboot to run cloud-init
+   from scratch:
+
+.. code:: bash
+
+    ssh ubuntu@$CI_VM_IP -- sudo hostname something-else
+    ssh ubuntu@$CI_VM_IP -- sudo cloud-init clean --logs --reboot
+
+5. Validate **-proposed** cloud-init came up without error
+
+.. code:: bash
+
+    # Block until cloud-init completes and verify from --long that the proper
+    # datasource was detected. Errors would show up in --long
+    ssh ubuntu@$CI_VM_IP -- cloud-init status --wait --long
+    # Make sure hostname was set properly to SRU-worked-<cloud name>
+    ssh ubuntu@$CI_VM_IP -- hostname
+    # Check for any errors or warnings in cloud-init logs.
+    # (This should produce no output if successful.)
+    ssh ubuntu@$CI_VM_IP -- grep Trace "/var/log/cloud-init*"
+
+6. If you encounter an error during SRU testing:
+
+   * Create a `new cloud-init bug`_ reporting the version of cloud-init
+     affected **and** ping upstream cloud-init on Freenode's
```

Comment:
I know I suggested this, but I'm having second thoughts about the "and" here. I worry that people might bounce off at this point when they realise that they have to set up a Launchpad account _and_ an IRC client to follow these instructions correctly.

What do you think?

Reply:
Sure, will revert. the benefit too of linking to webchat.freenode.net is that they don't have to setup an IRC client (as I figured that'd be a barrier too) :)